server/src/test/java/org/elasticsearch/cluster/node/DiscoveryNodeFiltersTests.java

```diff
@@ -295,6 +295,16 @@ public void testOnlyAttributeValueFilter() {
         assertThat(discoveryNodeFilters.isOnlyAttributeValueFilter(), is(discoveryNodeFilters.match(node)));
     }
 
+    public void testIpv6WithScopeIdFromSettingsMatchingIpv6WithCompressedScopeId() {
```

Comment:
I don't think this has anything to do with scope IDs? The scope ID is a suffix on the address of the form `%eth0`. How about `testNormalizesIPAddressFilters` (or `testNormalizesIPv6AddressFilters` and then a separate test for IPv4 addresses?